### ISSUE TITLE
docker compose up environment variables using a .env

### ISSUE THREAD

**Issue by monstrodopantano** [microsoft/vscode-docker#3266]
Hi All, 
I would like to report a problem with plugin in the following configuration.

Steps to Reproduce:

1. I'm use environment variables in shell to populate values inside a Compose file
- https://docs.docker.com/compose/environment-variables/
```yml
//docker-compose.dev.yml
version: "3.7"

services:
  php7_id354:
    build: ./myapp-backend
    ports:
        - ${PORT_PHP}:80
    volumes:
        - ${FRONTEND_PATH}:/var/www/html/myapp-front
        - ${BACKEND_PATH}:/var/www/html/myapp-backend

  composer_id354:
    image: composer:2.0.13
    volumes:
        - ${BACKEND_PATH}/composer.json:/app/composer.json
        - ${BACKEND_PATH}:/app
    command: bash -c "composer update --quiet --ignore-platform-reqs"

  node_id357:
    build: ./myapp-front
    ports:
        - ${PORT_NODE}:3000
    volumes:
        - ${FRONTEND_PATH}:/app/myapp-front
        - ${FRONTEND_PATH}/node_modules:/app/myapp-front/node_modules
```
2. My file /var/www/myapp-backend/.env
```txt
PORT_NODE=3001
FRONTEND_PATH=/var/www/myapp-front

PORT_PHP=8083
BACKEND_PATH=/var/www/myapp-backend
```
3. I'm use template to up my containers. I'm  can set default values for environment variables using a .env file, which Compose automatically looks for in project directory (parent folder of your Compose file). Values set in the shell environment override those set in the .env file.
- https://code.visualstudio.com/docs/containers/docker-compose#_template-matching
```yml
 "docker.commands.composeUp": [
        {
            "type": "shell",
            "label": "override",
            "template": "docker-compose --env-file ./myapp-backend/.env -f docker-compose.dev.yml up",
            "presentation": {
                "reveal": "always",
                "panel": "new"
            }
        }
    ]
```
4. Run command of template yaml of plugin directly in terminal I have success, the same command through the plugin does not. See error plugin.

```shell
> Executing task: docker-compose -f 'docker-compose.dev.yml' -p 'myapp' stop  <

ERROR: The Compose file './docker-compose.dev.yml' is invalid because:
services.node_id357.ports contains an invalid type, it should be a number, or an object
services.php7_id354.ports contains an invalid type, it should be a number, or an object
The terminal process "zsh '-c', 'docker-compose -f 'docker-compose.dev.yml' -p 'myapp' stop '" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```
- VS Code Version: 1.60.2
- Plugin Docker: 1.17.0
- OS Version: Debian 11

**Comment by bwateratmsft** [comment]
That makes sense. The customization for the `composeUp` command will not affect the start, stop, or restart commands. It seems unfortunate that Compose gets upset about unresolved environment variables in the config when they are not needed in order to stop the containers...

**Comment by bwateratmsft** [comment]
Took me a while to get to it but I did some investigating. The environment file used _is_ specified as a label in the compose'd containers, `com.docker.compose.project.environment_file`. Separately from that I noticed we totally messed up with taking an `envFiles` parameter to compose tasks, it only supports _one_ file. We need to adjust that and the documentation while doing our best to support backwards compatibility.
